### ISSUE TITLE
Default JSON.net configuration corrupts date strings when deserializing JSON data for the backoffice

### ISSUE THREAD

**Issue by abjerner** [umbraco/Umbraco-CMS#3075]
With the default configuration of JSON.net, whenever it deserializes a string into data, it will look for string properties, where the value looks to be in the format of a date. If this is the case, JSON.net will internally parse and store the value as an instance of `System.DateTime` instead of `System.String`.

As the value is being stored as an instance of `System.DateTime`, JSON.net has no recollection of the original format the date was formatted in, so when the same JSON data is later serialized, the saved value is similar to the one returned by calling the parameterless `ToString` method of `System.DateTime`. 

Using invariant culture (or `en-GB` or `da-DK`), a date like `2018-09-12T09:44:53.961+0000` will be converted to `12/09/2018 11:44:53`, but with `en-US` the same date will be converted to `9/12/2018 11:44:53 AM`. Either way, the format is different, which results in us loosing both the precision and the UTC offset - and it may even break our code if we're expecting a specific format.

### Steps to reproduce
Say we have a raw JSON string representing an object with a date, the code for saving that data could look like:

```csharp
// The original JSON string
string rawJson = "{\"date\":\"2018-09-12T09:44:53.961+0000\"}";

// Get a reference to the content item we wish to update
IContent content = ApplicationContext.Services.ContentService.GetById(1140);

// Set the property value
content.SetValue("data", rawJson);

// Save and publish
ApplicationContext.Services.ContentService.SaveAndPublishWithStatus(content);
```

The JSON string is correctly saved to the database, it has only been treated as a string at this point. However if a user visits the content item in the backoffice, the raw string is being deserialized, and if the JSON value is outputted, it will look like as shown below:

![image](https://user-images.githubusercontent.com/3634580/46251004-b4cf3280-c447-11e8-896c-2044ef75cc60.png)

`2018-09-12 11:44:53` is the result you get from calling `DateTime.Parse("2018-09-12T09:44:53.961+0000").ToString()`. I'm not sure whether this is done using the invariant culture or the default culture on my machine. I'm running the backoffice using `en-US`, which would have been `9/12/2018 11:44:53 AM` - so the current backoffice culture isn't used.

### Expected result

When saving a JSON data structure via the content service (of the backoffice for that matter), I would expect the same the date format to be persisted when later saving from the backoffice.

### Potential fix

JSON.net has a `DateParseHandling` option, where the default value is `DateParseHandling.DateTime`. The option can be set to `None`, in which case JSON.net doesn't try to convert strings to `System.DateTime`. See an example here:

https://github.com/skybrud/Skybrud.Essentials/blob/master/src/Skybrud.Essentials/Json/JsonUtils.cs#L28

I'm however not sure where to apply this change in the code 😮

**Comment by nul800sebastiaan** [comment]
Thanks @abjerner - I'll have to discuss this in our next regular meeting about incoming issues.

**Comment by nul800sebastiaan** [comment]
Talked to Shannon and his feedback was: 

- We can't really change this behavior now, this would change the behavior for existing property editors
- You can implement a bit of C# for your property editor overriding `ConvertDbToEditor` to use your own Json options
- You could hack it by wrapping the date in single quotes then our magic convertor doesn't kick in and will leave the value alone 🙈 - I wouldn't recommend it but it could help if you don't want to add a dll to your plugin

**Comment by abjerner** [comment]
@nul800sebastiaan I'm taking the liberty to re-open this issue, as it still exists. What is the best approach to make sure this is handled correctly for V10?

Given that I reported this back in 2018, I'm assuming it was for V7 - and it was then overlooked for both V8 and V9.

**Comment by Zeegaan** [comment]
@abjerner I suspect this is still an issue for v10+, could you come up with a new code example showing this? 🤔 
We're already changing the way objects a serialized/deserialized for v13, so I think if we need to make a change for this, it should also be now 😁

**Comment by abjerner** [comment]
@Zeegaan yes, it's still an issue. My snippet for reproducing the issue is roughly the same, except that you now need to inject `IContentService` rather than using `ApplicationContext.Services.ContentService`.

**Comment by Zeegaan** [comment]
@abjerner and what DataType are you using then 🤔

**Comment by abjerner** [comment]
@Zeegaan I'd say basically any data type that stores a JSON value and also allows dates as part of that data. So not any specific data type.

Our use case is typically that we create a custom property editor to store a JSON blob that we have from some external source. If that JSON blob contains dates recognized by JSON.net, JSON.net will parse them to `DateTime`, and then likely back a different format when the JSON value is again serialized.

Like Seb wrote, there are a few ways to get around this. I tend to prefix the raw JSON value with an underscore, as Umbraco/JSON.net then won't see the string value as a JSON string, and thus leave it alone. I can then parse the JSON on my own in my property editor's value converter.

**Comment by Zeegaan** [comment]
@abjerner If any could you just set up a simple example for me please 🙈 I feel a bit stupid, but I'm not following at all here 😅
